packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts

```diff
@@ -2062,6 +2060,65 @@ declare module 'lightning/analyticsWaveApi' {
         sourceObjectFields: Array<string>;
     }
 
+    /**
+     * An Analytics template asset reference.
+     *
+     * https://developer.salesforce.com/docs/atlas.en-us.bi_dev_guide_rest.meta/bi_dev_guide_rest/bi_responses_asset_reference.htm
+     *
+     * Keys:
+     *    (none)
+     */
+    export interface TemplateAssetReferenceRepresentation {
+        /** The 18 character ID of the asset. */
+        id?: string;
+        /** The asset label. */
+        label?: string;
+        /** The asset developer name. */
+        name?: string;
+        /** The namespace that qualifies the asset name */
+        namespace?: string;
+        /** The asset URL. */
+        url?: string;
+    }
+
+    /**
+     * Representation for individual validation task
+     *
+     * https://developer.salesforce.com/docs/atlas.en-us.bi_dev_guide_rest.meta/bi_dev_guide_rest/bi_responses_template_readiness_item.htm
+     *
+     * Keys:
+     *    (none)
+     */
+    export interface TemplateReadinessItemRepresentation {
+        /** The icon/image associated with the validation task. */
+        image: TemplateAssetReferenceRepresentation | null;
+        /** The task specific label. */
+        label: string | null;
+        /** The task specific message. */
+        message: string | null;
+        /** The status for the readiness task */
+        readinessStatus: string | null;
```

Comment:
Since this is like an enum, could it use string literals in the type, like `readinessStatus: "failed" | "success" | null` (just making up those values, not sure if they are the right ones)?

Reply:
These types come from the generated adapter code, and are basically copied here. And, we need to match what the adapter code says, since any usages of this in js will call through the adapter. In the adapter code, it (probably) intentionally doesn't pull union types for underlying enums, but just does `string`, so we do that here, too. We sometimes put a note in the jsdoc about the possible values, but generally just rely on the docs, since those get updated per release.